addons/binding/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/protocol/RemoteControllerWebSocket.java

```diff
@@ -0,0 +1,678 @@
+        for (App app : apps.values()) {
+            lastApp = app.appId;
+        }
+
+        for (App app : apps.values()) {
+            getAppStatus(app.appId);
+        }
+    }
+
+    /**
+     * Websocket class for remote control
+     *
+     * @author Arjan Mels
+     *
+     */
+    class WebSocketRemote extends WebSocketBase {
+        @Override
+        public void onWebSocketError(Throwable error) {
+            super.onWebSocketError(error);
+            callback.connectionError(error);
+        }
+
+        @Override
+        public void onWebSocketText(String msgarg) {
+            String msg = msgarg.replace('\n', ' ');
+            super.onWebSocketText(msg);
+            try {
+                JSONObject json = (JSONObject) new JSONParser(JSONParser.MODE_PERMISSIVE).parse(msg);
+                switch (json.getAsString("event")) {
+                    case "ms.channel.connect":
+                        logger.debug("Remote channel connected");
+                        getApps();
+                        break;
+                    case "ms.channel.clientConnect":
+                        logger.debug("Remote client connected");
+                        break;
+                    case "ms.channel.clientDisconnect":
+                        logger.debug("Remote client disconnected");
+                        break;
+                    case "ed.edenTV.update":
+                        logger.debug("edenTV update: {}", ((JSONObject) json.get("data")).getAsString("update_type"));
+                        updateCurrentApp();
+                        break;
+                    case "ed.apps.launch":
+                        logger.debug("App launched: {}",
+                                ((JSONObject) ((JSONObject) json.get("params")).get("data")).get("appId"));
+                        break;
+                    case "ed.installedApp.get":
+                        apps.clear();
+
+                        JSONObject data = (JSONObject) json.get("data");
+                        JSONArray array = (JSONArray) data.get("data");
+
+                        for (Object jsonApp : array) {
+                            App app = new App(((JSONObject) jsonApp).getAsString("appId"),
+                                    ((JSONObject) jsonApp).getAsString("name"),
+                                    ((JSONObject) jsonApp).getAsNumber("app_type").intValue());
+                            apps.put(app.name, app);
+                        }
+
+                        logger.debug("Installed Apps: "
+                                + apps.entrySet().stream().map(entry -> entry.getValue().appId + " = " + entry.getKey())
+                                        .collect(Collectors.joining(", ")));
+
+                        updateCurrentApp();
+
+                        break;
+                    default:
+                        logger.debug("WebSocketRemote Unknown event: {}", msg);
+
+                }
+            } catch (Exception e) {

```

Comment:
see above remark regarding `Exception`. You can also catch several excpetions in one catch-clause.

Reply:
Code changed, comment outdated